### PR TITLE
don't relay HTTP response codes in load balancing

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Don't relay HTTP/1.1 response headers in load-balanced environment.
This should fix the issue described in https://github.com/arangodb/arangojs/issues/655

- [x] Bug-Fix for *devel-branch* (i.e. no need for backports?)
- [ ] Bug-Fix for a *released version* (did you remember to port this to all relevant release branches?)
- [x] The behavior in this PR can be (and was) *manually tested* (support / qa / customers can test it)
- [x] The behavior change can be verified via automatic tests

#### Related Information

- [x] There is a GitHub Issue reported by a Community User: https://github.com/arangodb/arangojs/issues/655

### Testing & Verification

This PR adds tests that were used to verify all changes:

- [x] Added new **integration tests** (i.e. in scripts/unittest load_balancing)

http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/9354/

### CODE REVIEW

File: arangod/GeneralServer/RestHandler.cpp [hunk 1]
```diff
@@ -221,6 +221,11 @@ futures::Future<Result> RestHandler::forwardRequest(bool& forwarded) {
 
     auto const& resultHeaders = response.response->messageHeader().meta();
     for (auto const& it : resultHeaders) {
+      if (it.first == "http/1.1") {
```
Comment (goedderz) [inline arangod/GeneralServer/RestHandler.cpp:224]: I've no overview over the fuerte code. But shouldn't the status line be kept separately from the headers in the first place?

Comment (goedderz) [reply arangod/GeneralServer/RestHandler.cpp:224]: And is the string `http/1.1` future proof enough?

Comment (jsteemann) [reply arangod/GeneralServer/RestHandler.cpp:224]: @goedderz : I don't know. But I think for now it is safe.

Comment (graetzer) [reply arangod/GeneralServer/RestHandler.cpp:224]: its a hack to mimick some old behaviour of HttpClientConnection

Comment (goedderz) [reply arangod/GeneralServer/RestHandler.cpp:224]: 👍

**Pull request merged**